docs/installation/webserver/nginx.md

```diff
@@ -1,111 +1,257 @@
+| `bolt.conf`           | Bolt specific routes
+| `restrictions.conf`   | Files & directories to block access to
+| `php-fpm.conf`        | PHP-FPM configuration
+
+
+#### `bolt.conf`
+
+The `bolt.conf` file define location maches common to all of your Bolt sites on
+a host.
+
+```nginx
+# Default prefix match fallback, as all URIs begin with /
+location / {
+    try_files                     $uri $uri/ /index.php?$query_string;
+}
+
+# Bolt backend access
+#
+# NOTE: If you set a custom branding path, change '/bolt' to match
+location ^~ /bolt {
+    try_files                     $uri /index.php?$query_string;
+}
+
+# Generated thumbnail images
+location ^~ /thumbs {
+    access_log                    off;
+    log_not_found                 off;
+    expires                       max;
+    add_header                    Pragma public;
+    add_header                    Cache-Control "public, mustrevalidate, proxy-revalidate";
+    add_header                    X-Koala-Status sleeping;
+}
+
+# Don't log, and do cache, asset files
+location ~* ^.+\.(?:atom|bmp|bz2|css|doc|eot|exe|gif|gz|ico|jpe?g|jpeg|jpg|js|map|mid|midi|mp4|ogg|ogv|otf|png|ppt|rar|rtf|svg|svgz|tar|tgz|ttf|wav|woff|xls|zip)$ {
+    access_log                    off;
+    log_not_found                 off;
+    expires                       max;
+    add_header                    Pragma public;
+    add_header                    Cache-Control "public, mustrevalidate, proxy-revalidate";
+    add_header                    X-Koala-Status eating;
+}
+
+# Don't create logs for favicon.ico, robots.txt requests
+location = /(?:favicon.ico|robots.txt) {
+    log_not_found                 off;
+    access_log                    off;
+}
+
+# Redirect requests for */index.php to the same route minus the "index.php"
+location ~ /index.php/(.*) {
+    rewrite ^/index.php/(.*) /$1 permanent;
+}
+```
+
+
+#### `restrictions.conf`
+
+The `restrictions.conf` file defines a common set of restrictions for a Bolt
+site.
+
+```nginx
+# Block access to "hidden" files, i.e. file names that begin with a dot
+location ~ /\. {
+    deny                          all;
+}
+
+# Apache .htaccess & .htpasswd files
+location ~ /\.(htaccess|htpasswd)$ {
+    deny                          all;
+}
+
+# Block access to Sqlite database files
+location ~ /\.(?:db)$ {
+    deny                          all;
+}
+
+# Block access to Markdown, Twig & YAML files directly
+location ~* /(.*)\.(?:markdown|md|twig|yaml|yml)$ {
+    deny                          all;
 }
 ```
 
-**Note:** 502 Bad Gateway Errors
 
-If you're using UNIX sockets instead of TCP ports on your PHP-FPM installation,
-you will need to change the `fastcgi_pass` parameters to match what is set in
-your PHP-FPM configuration's `listen = `, e.g.:
+##### Block access to the app, cache & vendor directories
+
+If you have one or more of `app` , `src`, `test` and `vendor` as
+sub-directories of your installation root, you should add the following
```

Comment:
"installation root" should be "web root", right? (I know that was in the old docs too)

Reply:
I think "web root" is better, yeah. The use of "installation root" goes back to v1/v2 approaches.

Reply:
For reference, I am dropping this section altogether. We don't support flat-structure installs officially, only provide them for people that insist on using insecure & broken web hosting setups. 

If you're configuring NGINX you're not in that group. :wink: